src/main/java/org/candlepin/hostedtest/HostedTestCloudRegistrationAdapter.java

```diff
@@ -129,6 +130,10 @@ public CloudAuthenticationResult resolveCloudRegistrationDataV2(CloudRegistratio
             throw new MalformedCloudRegistrationException(
                 "missing offer ID in registration information metadata");
         }
+        else if (offerId.equals("1POffering")) {
+            throw new CloudRegistrationNotSupportedForOfferingException(
+                "cloud registration v2 is not supported for 1P offerings");
+        }
```

Comment:
The magic string scares me a bit. I wonder if we should add a offer type field to the body of the `POST cloud/offers` hosted test endpoint and store the type in the `HostedTestDataStore`.  Then we can check the type when we retrieve the product ID based on the offer ID and then throw the `CloudRegistrationNotSupportedForOfferingException` when the type is 1P.

Reply:
I'm undecided if that is needed, or just overcomplicates things. The `HostedTestDataStore` and `HostedTestResource` in general are test code, and the only reason we structure data (like products, subscriptions, etc.) is because those are also reflected on internal candlepin code. While in this case, Candlepin won't be dealing with an offering/product/subscription/pool '1P' type. This piece of data is entirely transparent to candlepin (adapters throw an exception, candlepin throws it further on).

It is not like we'll ever need to add more test scenarios with other 1P offerings

Reply:
Updated. See if this is the kind of thing you meant